enhancements/machine-api/opt-out-machine-health-checking.md

```diff
@@ -0,0 +1,378 @@
+instead of having a single blanket policy.
+
+## Related Information
+
+### What do Machines/MachineSets in a default OpenShift cluster look like?
+
+By default, in an OpenShift IPI installation, a MachineSet is created for each availability zone in the region that the
+cluster is deployed into, eg, in AWS US-East-1, 6 MachineSets are created.
+The installer will set the replica count to 1 for the first 3 MachineSets it creates and zero otherwise.
+
+These MachineSets are intended to create generic worker Machines are are all identical apart from the availability
+zone in which they are deployed.
+
+Each of theses MachineSets will apply 4 labels to the Machines that it creates:
+```yaml
+machine.openshift.io/cluster-api-cluster: <cluster-name>
+machine.openshift.io/cluster-api-machine-role: worker
+machine.openshift.io/cluster-api-machine-type: worker
+machine.openshift.io/cluster-api-machineset: <cluster-name>-worker-<availability-zone>
+```
+
+In addition to any Machines created by the MachineSets, the installer creates 3 Control-Plane Machines.
+This means that typically a cluster will have 6 Machines present at time of install.
+
+The Control-Plane machines are labelled:
+```yaml
+machine.openshift.io/cluster-api-cluster: <cluster-name>
+machine.openshift.io/cluster-api-machine-role: master
+machine.openshift.io/cluster-api-machine-type: master
+```
+
+Additionally all Machines are labelled with the following three labels:
+```yaml
+machine.openshift.io/instance-type: <instance-type>
+machine.openshift.io/region: <availability-zone>
+machine.openshift.io/zone: <availability-zone>
+```
+
+The labels applied to Machines are important as they allow users to determine which Machines are targeted by a Machine Health Check.
+
+### How does a Machine Health Check determine which Machines it should target for health checking?
+
+A MachineHealthCheck uses a `selector` to specify which Machines it should consider as its targets.
+The selector supports [set-based requirements](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements)
+so there are three options for specifying a selector for a Machine Health Check.
+
+#### An Empty Selector
+
+An empty selector matches all Machines. This means it would include the Control-Plane Machines,
+all Machines created by the default MachineSets and any Machines that users create post installation.
+
+#### Using exact labels
+
+Using the `matchLabels` part of the selector allows specifying a list of key value pairs to check the presence of.
+This means that Machines must have all of the labels specified to be considered to match the selector.
+This could be used to specify that the MHC selects all Machines with the `cluster-api-machine-role: worker` label,
+which would match all MachineSets created by the installer, but not Control-Plane nodes,
+nor necessarily any MachineSets created by users post installation.
```

Comment:
when you say

> nor necessarily any MachineSets created by users post installation.

is this referring to the idea that a user could remove `cluster-api-machine-role: worker`? (i had thought this label was applied by machine api)

Reply:
Users could remove that label from the pre-existing MachineSets created by the installer, or they could create new MachineSets which don't have that label on them. The label itself is copied from the MachineSet template onto the newly created Machines, there's nothing however to stop a user removing that